sql/src/main/java/org/apache/druid/sql/calcite/planner/PlannerContext.java

```diff
@@ -527,13 +522,9 @@ public boolean featureAvailable(final EngineFeature feature)
       // Short-circuit: feature requires context flag.
       return false;
     }
-
-    if (feature == EngineFeature.UNNEST &&
-        !QueryContexts.getAsBoolean(CTX_ENABLE_UNNEST, queryContext.get(CTX_ENABLE_UNNEST), false)) {
-      // Short-circuit: feature requires context flag.
+    if (engine == null) {
```

Comment:
Is this required anywhere in the main code? 

I think `EngineFeature` is `null` in one of the test cases. We should provide any implementation in that test case instead of making the change in the main code. This can hide bugs, if we unintentionally pass EngineFeature as `null` in the main code.

Reply:
Yes there is a `CalcitePlannerModuleTest` that was using an injector which comes into this case of the engine being a null. I'll check if I can handle it there with a mock. Can you please review the rest while I patch that up